src/components/Text/LinkDetectionText.tsx

```diff
@@ -9,30 +11,57 @@ export type LinkDetectionTextProps = {
   linkTextStyle?: StyleProp<TextStyle>;
   textStyle?: StyleProp<TextStyle>;
   parentTextStyle?: StyleProp<TextStyle>;
+  withTopicDetection?: boolean;
 };
 
 const LinkDetectionText = ({
   text,
   linkTextStyle = {},
   parentTextStyle = {},
-  textStyle = null
+  textStyle = null,
+  withTopicDetection = false
 }: LinkDetectionTextProps) => {
+  const navigation = useNavigation();
+
   const sanitizedTextPerLine = text?.split('\n');
 
+  const handleTopicPress = (topic: string) => {
+    const navigationParam = {
+      id: topic
+    };
+
+    navigation.navigate('TopicPageScreen', navigationParam);
+  };
+
   if (!textStyle) textStyle = linkTextStyle;
 
   return (
-    <Text style={[parentTextStyle]}>
+    <Text style={parentTextStyle}>
       {sanitizedTextPerLine?.map((line, index) => {
         if (index < sanitizedTextPerLine?.length - 1) {
           line += '\n';
         }
 
-        if (isValidUrl(line)) return <LinkableText style={[linkTextStyle]} text={line} />;
+        if (isValidUrl(line))
+          return (
+            <LinkableText
+              style={linkTextStyle}
+              text={line}
+              withTopicDetection={withTopicDetection}
+              isFirstLine={index === 0}
+            />
+          );
 
         return (
           <Text key={index} style={[textStyle]}>
-            {line}
+            {withTopicDetection && (
+              <TopicDetectionOnChat
+                text={line}
+                onPress={handleTopicPress}
+                isFirstLine={index === 0}
+              />
+            )}
+            {!withTopicDetection && line}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The rendering logic for different text parts has been refactored. If `withTopicDetection` is true, the `TopicDetectionOnChat` component is rendered for each line of text. This component detects topics in the chat text and allows for topic-specific onPress actions. If `withTopicDetection` is false, the line of text is rendered as it is.
```diff
+ return (
+   <Text style={parentTextStyle}>
+      {sanitizedTextPerLine?.map((line, index) => {
+        if (index < sanitizedTextPerLine?.length - 1) {
+          line += '\n';
+        }
+
+       if (isValidUrl(line))
+         return (
+           <LinkableText
+             style={linkTextStyle}
+             text={line}
+             withTopicDetection={withTopicDetection}
+             isFirstLine={index === 0}
+           />
+         );
+
+        return (
+          <Text key={index} style={[textStyle]}>
+           {withTopicDetection && (
+             <TopicDetectionOnChat
+               text={line}
+               onPress={handleTopicPress}
+               isFirstLine={index === 0}
+             />
+           )}
+           {!withTopicDetection && line}
+         </Text>
+       );
+     })}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->